src/pkg/cluster/state_test.go

```diff
@@ -5,16 +5,183 @@
 package cluster
 
 import (
+	"context"
 	"fmt"
 	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
 
 	"github.com/defenseunicorns/pkg/helpers"
+
+	"github.com/defenseunicorns/zarf/src/pkg/k8s"
 	"github.com/defenseunicorns/zarf/src/pkg/message"
 	"github.com/defenseunicorns/zarf/src/pkg/pki"
 	"github.com/defenseunicorns/zarf/src/types"
-	"github.com/stretchr/testify/require"
 )
 
+func TestZarfInitState(t *testing.T) {
+	t.Parallel()
```

Comment:
Echoing what @AustinAbro321 already stated about a race condition causing a nil panic due to the spinner not being goroutine safe

```bash
go test ./src/pkg/cluster -v -count=1 -failfast -race -run TestZarfInitState
```

<img width="1002" alt="data-race" src="https://github.com/defenseunicorns/zarf/assets/87675701/f0b62938-c8b5-4a5b-bae3-4be540bdc580">

It seems there may be other race condition bugs in our unit tests related to the spinner. To avoid this issue for now, I'm okay with _not_ running the test cases for `InitZarfState` in parallel. It would be helpful to leave a comment explaining why these tests cannot be run in parallel and to create a tech debt issue to address the spinner race condition. We should also consider adding the `-race` flag to our `go test` command in our `test-unit` make target to catch these problems in the future.